FHIRServicesExternalModule.php

```diff
@@ -734,7 +734,9 @@ function toArray($fhirObjectOrArray){
             $handle($a);
             return $a;
         }
-        else{
+        else if(is_null($fhirObjectOrArray)){
+            return "null";
+        } else {
```

Comment:
`null` is not a valid `$fhirObjectOrArray` or Array.  Instead of this check, would it be more appropriate to check for `null` before the `toArray()` function is called?

Reply:
I agree that the check can be moved to somewhere else.
The obvious option would be to move it the the caller `jsonSerialize`, since this function is the only place where `toArray` is called as far a I see.
Even though, the `jsonSerialize` function parameter is called `$fhirObjectOrArray` as well. Should we rename this parameter for maintenance/readability reasons to something like `$fhirObjectOrArrayOrNull`?
Moving the null-check even before the `jsonSerialize` is not useful in my opinion since the the function is called from nine different locations.
But, of course, I am interested in your opinion.

Reply:
I'm thinking about this a little differently.  The only case I know of where the value passed to `jsonSerialize()` or `toArray()` is expected to be `null` is on the `$response` in `send-record.php`.  In that light, I think it makes sense to put the `null` check in `send-record.php`, and leave the existing behavior everywhere else.  I believe [failing fast](https://en.wikipedia.org/wiki/Fail-fast) and crashing on `null` as soon as it is unexpected is a good thing.

Reply:
Ah, I see. Yes, that approach makes sense as well.
I changed the code in the request where the check is performed in `send-record.php`.

Reply:
The latest changes look good to me.  Resolving!